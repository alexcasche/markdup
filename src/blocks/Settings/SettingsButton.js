import { ButtonLink } from 'elements'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { shade, tint } from 'polished'

const MODIFIERS = {
  disabled: ({ theme }) => `
    border-radius: 4px 0 0 4px;
    pointer-events: none;
    opacity: 0.85;
  `,
}

const SettingsButton = ButtonLink.extend`
  width: 50%;
  margin-top: 15px;
  height: 40px;
  background-color: ${props => shade(0.75, props.theme.colors.blue)};
  border-radius: 0 4px 4px 0;
  color: ${props => tint(0.15, props.theme.colors.grey)};
  font-size: 14px;
  &:hover {
    background-color: ${props => shade(0.85, props.theme.colors.blue)};
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default SettingsButton
