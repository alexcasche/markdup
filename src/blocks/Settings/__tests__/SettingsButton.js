import SettingsButton from '../SettingsButton'

describe('SettingsButton Element', () => {
  it('renders correctly', () => {
    const tree = render(<SettingsButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds checked modifier', () => {
    const tree = render(<SettingsButton modifiers={['disabled']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('pointer-events', 'none')
  })
})
