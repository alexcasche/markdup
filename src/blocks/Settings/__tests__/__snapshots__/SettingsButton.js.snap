// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsButton Element adds checked modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  width: 50%;
  margin-top: 15px;
  height: 40px;
  background-color: #2b99b8;
  border-radius: 0 4px 4px 0;
  color: #f2f2f3;
  font-size: 14px;
  border-radius: 4px 0 0 4px;
  pointer-events: none;
  opacity: 0.85;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover {
  background-color: #31aed1;
}

<a
  className="c0"
/>
`;

exports[`SettingsButton Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  width: 50%;
  margin-top: 15px;
  height: 40px;
  background-color: #2b99b8;
  border-radius: 0 4px 4px 0;
  color: #f2f2f3;
  font-size: 14px;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover {
  background-color: #31aed1;
}

<a
  className="c0"
/>
`;
